Clinical-Genomics/scout
# Phenomodels lock 'n fold

This PR adds 2 functionalities:
- Lock phenomodels so that they are editable by one or more selected user collaborators --> fix #2236 
- Make the phenomodel checkboxes which have children term foldable (folded since the beginning) --> fix #2237 

**How to test --> local instance of Scout**:
1. create a new user for cust000:
```scout --demo load user -i cust000 -u "Diana Prince" -m wonderwoman@mail.com```
1. Log in as demo user Clark Kent
1. Go to the phenomodels page and create 3 phenomodels with submodels
1. Lock one model with admins Clark Kent and Diana Prince. (model 1)
1. Lock one model with admin Clark Kent (model 2)
1. Don't lock a model at all. (model 3)
1. Log out and log in with user wonderwoman@mail.com
1. Make sure that this user can access all 3 models but can modify only model 1 and 3, including changing the admin and locking these models.
1. Make sure that the phenomodel preview contains foldable divs for the highest level checkboxes which contain children terms (fold and unfold by clicking on their names):
![image](https://user-images.githubusercontent.com/28093618/102505024-cd0be600-4081-11eb-8fe2-04869bcf1556.png)



**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
